### DISCUSSION TITLE
Does any of  Anomalib library models support Anamoly Detection in 1D signal?

### DISCUSSION THREAD

**Discussion by bhargav-minto** [open-edge-platform/anomalib#918] [Q&A]
Hello everyone,

I'm working on a project where I want to detect anomalies in a 1D signal that represents spectrum data in the frequency domain. I'm wondering if any of the models in the Anamalib library are suitable for this task.

Specifically, I'm looking for a model that can work with 1D Spectrum and output a classification indicating whether the signal contains an anomaly or not.

 I would greatly appreciate any recommendations or advice.

Thank you!

**Comment by djdameln** [comment]
Unfortunately, Anomalib does not support 1D anomaly detection at this moment. We are planning to add this functionality in the future, but we are not yet able to provide an ETA for this. One possibility would be to convert your data to 2D spectrograms and process those with Anomalib's models. Of course, without knowing the further details of your project/dataset, it is hard to judge if this would be a suitable approach for your problem. Maybe you could have a look at [this discussion thread](https://github.com/openvinotoolkit/anomalib/discussions/798) where I answered a similar question. If you have any additional questions please do not hesitate to reach out!

**Comment by tominator95** [comment]
I had a project some time ago regarding classification of audio signals. I converted the 1d waveform to mel-spectograms via fourier transformation. With the normal and abnormal images I trained a image classification network. But as @djdameln mentioned, it would also be interesting to try it with some anomaly detection algorithms provided by Anomalib. Now I have a new project idea, thanks 👍
